doc: ironside: add note on placement of update (#24874)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_ironside.rst

|ISE| exposes an update service that allows local domains to trigger the update process by indirectly writing to the relevant SICR registers.

> .. note::
>    The update data needs to be placed within a valid memory range.
>    See ``nrf_ironside/update.h`` for more details.
> 
> 

Reviewer: ```suggestion
.. note::
   The update data must be placed within a valid memory range.
   See :file:`nrf_ironside/update.h` for more details.

```
